tests/kernel/sched/schedule_api/src/test_sched_pending.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <ksched.h>
+#include <spinlock.h>
+#include <wait_q.h>
+#include <ztest.h>
+
+struct sched_pending { 
+	_wait_q_t waitq;
+	struct k_spinlock lock;
+}pending;
+
+#define STACK_SIZE	(1024 + CONFIG_TEST_EXTRA_STACKSIZE)
+
+K_THREAD_STACK_DEFINE(test_stack, STACK_SIZE);
+
+static struct k_thread tdata;
+static int num;
+static k_tid_t tid;
+
+/* action for self-defined assert handler. */
+static ZTEST_DMEM volatile bool valid_assert;
+
+static inline void set_assert_valid(bool valid)
+{
+	valid_assert = valid;
+}
+
+static void action_after_assert_fail(void)
+{
+	k_thread_abort(tid);
+	
+	ztest_test_pass();
+}
+
+#ifdef CONFIG_ASSERT_NO_FILE_INFO
+void assert_post_action(void)
+#else
+void assert_post_action(const char *file, unsigned int line)
```

Comment:
The common fatal and assert handler has been merged, could you please leverage them instead? thanks!  
please refer to https://github.com/zephyrproject-rtos/zephyr/pull/30150